packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1540,4 +1541,32 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
 
     return getOptionStyles(this.props.theme!, customStylesForAllOptions, customStylesForCurrentOption);
   }
+
+  /**
+   * Get the aria-activedescendant value for the comboxbox.
+   * @returns the id of the current focused combo item, otherwise the id of the currently selected element, null otherwise
+   */
+  private _getAriaActiveDescentValue(): string | null {
+    let descendantText = (this.state.isOpen && (this.state.selectedIndex as number) >= 0 ? (this._id + '-list' + this.state.selectedIndex) : null);
+    if (this.state.isOpen && this.state.focused && this.state.currentPendingValueValidIndex !== -1) {
+      descendantText = (this._id + '-list' + this.state.currentPendingValueValidIndex);
+    }
+    return descendantText;
+  }
+
+  /**
+  * Get the aria autocomplete value for the Combobox
+  * @returns 'inline' if auto-complete automatically dynamic, 'both' if we have a list of possible values to pick from and can
+  * dynamically populate input, and 'none' if auto-complete is not enabled as we can't give user inputs.
+  */
+  private _getAriaAutoCompleteValue(): string {
+    let autoComplete = !this.props.disabled && this.props.autoComplete === 'on';
+    if (autoComplete && this.props.allowFreeform) {
+      return 'inline';
+    } else if (autoComplete && !this.props.allowFreeform) {
+      return 'both';
+    } else {
+      return 'none';
+    }
```

Comment:
Consider changing this to a ternary opertation like: `autoComplete ? allFreeform ? 'inline' : 'both' : 'none'` or `autoComplete ? (allFreeform ? 'inline' : 'both') : 'none'` (if you want to be explicit about it). This will make the statements cleaner and remove a duplicate autoComplete check

Reply:
Will do. I was once told to avoid it, but if you think it's fine, then I'll make the change.

Reply:
Yeah if it's a complicated ternary statement or you're nesting more than two statements, is how I usually think about it